server/src/middleware/errorHandler.ts

```diff
@@ -1,9 +1,46 @@
 import { Request, Response, NextFunction } from "express";
-import { BaseError, NotFoundError, ValidationError } from "../Errors/errors.js";
+
+// base error class
+export class BaseError extends Error {
+    status: boolean;
+    statusCode: number;
+
+    constructor(message: string, status: boolean, statusCode: number) {
+        super(message);
+        this.status = status;
+        this.statusCode = statusCode;
+        Object.setPrototypeOf(this, BaseError.prototype);
+    }
+}
+
+// 404 error class
+export class NotFoundError extends BaseError {          // missing user input
+    constructor(message: string) {
+        super(message, false, 404);
+        Object.setPrototypeOf(this, NotFoundError.prototype);
+    }
+}
+// Unauthorized error class
+export class UnAuthorizedError extends BaseError {          // missing user input
+    constructor(message: string) {
+        super(message, false, 401);
+        Object.setPrototypeOf(this, UnAuthorizedError.prototype);
+    }
+}
+
+// validation error class
+export class ValidationError extends BaseError {
+    errorData: Record<string, string>
+    constructor(data: Record<string, string>) {
+        super("Validation Error", false, 400);
+        this.errorData = data;
+        Object.setPrototypeOf(this, ValidationError.prototype)
+    }
+}
 
 // error handler middleware
 export const errorHandler: any = (err: Error, req: Request, res: Response, next: NextFunction) => {
-    console.log("error:", err)
+    console.log("error::", err)
```

Comment:
remove console.log

Reply:
This console is kept for bugging purpose for any kind of error. I will remove it at the the end.